src/main/java/com/squareup/javapoet/ClassName.java

```diff
@@ -44,7 +44,7 @@ private ClassName(List<String> names) {
       checkArgument(SourceVersion.isName(names.get(i)), "part '%s' is keyword", names.get(i));
     }
     this.names = Util.immutableList(names);
-    this.canonicalName = names.get(0).isEmpty()
+    this.canonicalName = (names.get(0) == null || names.get(0).isEmpty())
```

Comment:
Eep, does `null` actually happen? We should figure out where that's coming from and replace it with `""`.